lib/Resizable.js

```diff
@@ -4,227 +4,492 @@ import PropTypes from 'prop-types';
+                return this.resizeFromBottomLeftCorner(x, y, canDragY, deltaY,deltaX)
+            case 'react-resizable-handle':
+            case 'react-se-resizable-handle':
+                return this.resizeFromBottomRightCorner(deltaY, deltaX, canDragY, canDragX)
+            default :
+                return null
+        }
+    }
+
+    onResize = (event, {node, deltaX, deltaY, x, y}) => {
+        const canDragX = this.props.axis === 'both' || this.props.axis === 'x';
+        const canDragY = this.props.axis === 'both' || this.props.axis === 'y';
+
+        let width = this.state.width
+        let height = this.state.height
+        let left = this.state.left
+        let top = this.state.top
+
+        if (this.isResizeHandle(this.state.resizableHandle)) {
+            const dimension = this.getDimensionAcordingToMousePosition(this.state.resizableHandle,
+                x, y, canDragX, canDragY, deltaX, deltaY)
+
+            width = dimension.width
+            height = dimension.height
+            top = dimension.top
+            left = dimension.left
+        }
+
+        [width, height] = this.runConstraints(width, height)
+
+        if (!this.hasDimensionChanged(width, height, left, top)) {
+            return;
+        }
+
+        const resizeState = this.onResizeState(width,height,left,top)
+
+        if (!this.onResize) {
+            this.setState(resizeState);
+            return
+        }
+
+        this.persistEventIfFunction(event)
+
+        this.setState(resizeState, () =>
+            this.props.onResize(event,
+                {
+                    node,
+                    size: this.getDimension(resizeState)
+                }));
+
+    }
+
+    onResizeStop = (event, {node}) => {
+        if (!this.props.onResizeStop) {
+            this.setState(this.onResizeStopState());
+            return
+        }
+
+        this.persistEventIfFunction(event)
+
+        this.setState(this.onResizeStopState(), () =>
+            this.props.onResizeStop(event,
+                {
+                    node,
+                    size: this.getDimension(this.state)
+                })
+        );
+    }
+
+    onResizeStart = (event, {node, x, y}) => {
+
+        if (this.isResizeHandle(event.target.className)) {
+            this.setState({resizableHandle: event.target.className})
+        }
+
+        const resizeStartState = this.onResizeStartState(x, y)
+
+        if (!this.props.onResizeStart) {
+            this.setState(resizeStartState);
+            return
+        }
+
+        this.persistEventIfFunction(event)
+
+        this.setState(resizeStartState, () =>
+            this.props.onResizeStart(event,
+                {
+                    node,
+                    size: this.getDimension(this.state)
+                }
+            ));
+    }
+
+    getDimension = (state) => (
+        {width: state.width, height: state.height, left: state.left, top: state.top}
+    )
```

Comment:
You can simplify it by: 
```
getDimension = ({width, height, left, top}) =>
        ({width, height, left, top})
```
Note, the convention is to place '(' and ')' like that, please update other places as well.

Reply:
done